src/renderer/components/ConfigureServer.tsx

```diff
@@ -151,8 +164,34 @@ function ConfigureServer({
         setUrl(value);
 
         if (urlError) {
-            setURLError('');
+            setURLError(undefined);
         }
+
+        clearTimeout(validationTimeout.current as unknown as number);
+        validationTimeout.current = setTimeout(() => {
+            const currentTimeout = validationTimeout;
+            setValidating(true);
+            setURLError({
+                type: STATUS.INFO,
+                value: formatMessage({id: 'renderer.components.configureServer.url.validating', defaultMessage: 'Validating...'}),
+            });
+            validateURL(value).then(({validatedURL, serverName, message}) => {
+                if (currentTimeout.current !== validationTimeout.current) {
+                    return;
+                }
+                if (validatedURL) {
+                    setUrl(validatedURL);
+                }
+                if (serverName && !name) {
```

Comment:
The value of `name` here may be from a few renders before, so may not be up to date. 0/5 on why this check is here, so maybe it is not an issue, but if you foresee this being an issue, I would suggest using the "functional form" of the set state.

```
if (serverName) {
    setName((prev) => prev ? prev : serverName);
}
```